common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -916,97 +905,23 @@ private static boolean isWindows0() {
-            try {
-                p = Runtime.getRuntime().exec(new String[] { idCmd, "-u" });
-                in = new BufferedReader(new InputStreamReader(p.getInputStream(), CharsetUtil.US_ASCII));
-                uid = in.readLine();
-                in.close();
-
-                for (;;) {
-                    try {
-                        int exitCode = p.waitFor();
-                        if (exitCode != 0) {
-                            uid = null;
-                        }
-                        break;
-                    } catch (InterruptedException e) {
-                        // Ignore
-                    }
-                }
-            } catch (Throwable ignored) {
-                // Failed to run the command.
-                uid = null;
-            } finally {
-                if (in != null) {
-                    try {
-                        in.close();
-                    } catch (IOException e) {
-                        // Ignore
-                    }
-                }
-                if (p != null) {
-                    try {
-                        p.destroy();
-                    } catch (Exception e) {
-                        // Android sometimes triggers an ErrnoException.
-                    }
-                }
-            }
-
-            if (uid != null && UID_PATTERN.matcher(uid).matches()) {
-                logger.debug("UID: {}", uid);
-                return "0".equals(uid);
-            }
-        }
-
-        logger.debug("Could not determine the current UID using /usr/bin/id; attempting to bind at privileged ports.");
-
-        Pattern PERMISSION_DENIED = Pattern.compile(".*(?:denied|not.*permitted).*");
-        for (int i = 1023; i > 0; i --) {
-            ServerSocket ss = null;
-            try {
-                ss = new ServerSocket();
-                ss.setReuseAddress(true);
-                ss.bind(new InetSocketAddress(i));
-                if (logger.isDebugEnabled()) {
-                    logger.debug("UID: 0 (succeded to bind at port {})", i);
-                }
-                return true;
-            } catch (Exception e) {
-                // Failed to bind.
-                // Check the error message so that we don't always need to bind 1023 times.
-                String message = e.getMessage();
-                if (message == null) {
-                    message = "";
-                }
-                message = message.toLowerCase();
-                if (PERMISSION_DENIED.matcher(message).matches()) {
-                    break;
-                }
-            } finally {
-                if (ss != null) {
-                    try {
-                        ss.close();
-                    } catch (Exception e) {
-                        // Ignore.
-                    }
+    private static boolean maybeSuperUser0() {
+        try {
+            final String username = AccessController.doPrivileged(new PrivilegedAction<String>() {
```

Comment:
last comment. we shouldn't have to catch anything in this method after this change, right?